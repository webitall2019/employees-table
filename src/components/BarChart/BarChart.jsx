import React from "react";
import { map, get, toArray, countBy } from "lodash";
import "chart.js/auto";
import { Bar } from "react-chartjs-2";
import { useSelector } from "react-redux";

const BarChart = () => {
  const users = useSelector((state) => state);

  const sex = [];

  map(users, (man) => sex.push(get(man, ["gender"], "")));

  const uniqSexValues = new Set(sex);
  const labels = toArray(uniqSexValues);

  const options = {
    responsive: true,
    plugins: {
      legend: {
        display: true,
        position: "bottom",
      },
      title: {
        display: true,
      },
    },
  };
  const data = {
    labels: " ",
    datasets: [
      {
        label: "Male",
        data: labels.map(() => countBy(sex).Male),
        backgroundColor: "rgba(255, 99, 132, 0.5)",
      },
      {
        label: "Female",
        data: labels.map(() => countBy(sex).Female),
        backgroundColor: "rgba(53, 162, 235, 0.5)",
      },
    ],
  };
  return (
    <div className="bar-chart chart" data-testid="barChart">
      <Bar options={options} data={data} />
    </div>
  );
};

export default BarChart;
