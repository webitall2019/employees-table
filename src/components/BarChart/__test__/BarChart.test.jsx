import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import store from "../../../store/store";
import BarChart from "../BarChart";

describe("<BarChart/> component ", () => {
  it("check to be in the document", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BarChart />
      </Provider>
    );
    const barChartComponent = screen.getByTestId("barChart");
    expect(barChartComponent).toBeInTheDocument();
  });
  it("should match snapshot WHEN rendered", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BarChart />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
