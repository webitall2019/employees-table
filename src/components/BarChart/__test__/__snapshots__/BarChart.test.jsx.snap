// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChart/> component  should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="bar-chart chart"
    data-testid="barChart"
  />
</DocumentFragment>
`;
